From 351df2a411366767ad8759344fc255a6f22cd21b Mon Sep 17 00:00:00 2001
From: Matthias Kaehlcke <mka@chromium.org>
Date: Tue, 25 Sep 2018 15:22:03 -0700
Subject: [PATCH] CHROMIUM: Revert "Makefile: disable PIE before testing asm
 goto"

This reverts commit 8438ee76b004ef66d125ade64c91fc128047d244.

'asm goto' is currently not supported by clang.

BUG=chromium:891028
TEST=build for caroline

Change-Id: Iab421ebdb58ab5063e1f2b3f9553c0a347bd76b6
Signed-off-by: Matthias Kaehlcke <mka@chromium.org>

[rebase419(rrangel): Context conflict]
Signed-off-by: Raul E Rangel <rrangel@chromium.org>
---
 Makefile | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 9b4fa310b63a..3a655ff20b26 100644
--- a/Makefile
+++ b/Makefile
@@ -505,9 +505,6 @@ RETPOLINE_VDSO_CFLAGS := $(call cc-option,$(RETPOLINE_VDSO_CFLAGS_GCC),$(call cc
 export RETPOLINE_CFLAGS
 export RETPOLINE_VDSO_CFLAGS
 
-KBUILD_CFLAGS	+= $(call cc-option,-fno-PIE)
-KBUILD_AFLAGS	+= $(call cc-option,-fno-PIE)
-
 # The expansion should be delayed until arch/$(SRCARCH)/Makefile is included.
 # Some architectures define CROSS_COMPILE in arch/$(SRCARCH)/Makefile.
 # CC_VERSION_TEXT is referenced from Kconfig (so it needs export),
@@ -594,6 +591,8 @@ endif
 # Defaults to vmlinux, but the arch makefile usually adds further targets
 all: vmlinux
 
+KBUILD_CFLAGS	+= $(call cc-option,-fno-PIE)
+KBUILD_AFLAGS	+= $(call cc-option,-fno-PIE)
 CFLAGS_GCOV	:= -fprofile-arcs -ftest-coverage \
 	$(call cc-option,-fno-tree-loop-im) \
 	$(call cc-disable-warning,maybe-uninitialized,)
-- 
2.23.0.rc1.153.gdeed80330f-goog

